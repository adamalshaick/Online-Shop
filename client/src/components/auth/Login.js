import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { loginUser } from "../../actions/authActions";
import { Link } from "react-router-dom";
import TextFieldGroup from "../common/TextFieldGroup";

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: "",
      errors: {}
    };
  }

  componentDidMount() {
    if (this.props.auth.isAuthenticated) {
      this.props.history.push("/dashboard");
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.auth.isAuthenticated) {
      this.props.history.push("/dashboard");
    }

    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }
  }

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  onSubmit = e => {
    e.preventDefault();
    const userData = {
      email: this.state.email,
      password: this.state.password
    };
    this.props.loginUser(userData);
  };
  render() {
    const { errors } = this.state;

    return (
      <>
        <div className="row mt-5">
          <div className="col-md-2 col-lg-3" />
          <div className="col-md-8 col-lg-6 card mt-5 text-center p-0 entry">
            <div className="card-body p-5">
              <h3 className="mb-5">Log in to your account</h3>
              <form onSubmit={this.onSubmit}>
                <TextFieldGroup
                  id="#email"
                  placeholder="Email Adress"
                  name="email"
                  type="email"
                  value={this.state.email}
                  onChange={this.onChange}
                  error={errors.email}
                  id="email"
                />

                <TextFieldGroup
                  id="#password"
                  placeholder="Password"
                  name="password"
                  type="password"
                  value={this.state.password}
                  onChange={this.onChange}
                  error={errors.password}
                  id="password"
                />

                <button type="submit" className="btn btn-dark btn-block mt-4">
                  Log In
                </button>
              </form>
            </div>
            <div className="text-muted card-footer mt-3">
              Don't have an account yet? <Link to="/register">Sign Up</Link>
            </div>
          </div>
        </div>
      </>
    );
  }
}

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { loginUser }
)(Login);
