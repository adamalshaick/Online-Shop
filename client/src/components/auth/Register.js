import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { registerUser } from "../../actions/authActions";
import { Link } from "react-router-dom";
import TextFieldGroup from "../common/TextFieldGroup";
import Navbar from "../layout/Navbar";
import InputGroup from "../common/InputGroup";
import handleInputErrors from "../common/hoc/handleInputErrors";
import redirectAuthenticated from "../common/hoc/redirectAuthenticated";

export class Register extends Component {
  constructor() {
    super();
    this.state = {
      name: "",
      email: "",
      password: "",
      password2: "",
      bio: "",
      location: ""
    };
  }

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = e => {
    e.preventDefault();

    const newUser = {
      name: this.state.name,
      email: this.state.email,
      password: this.state.password,
      password2: this.state.password2,
      location: this.state.location,
      bio: this.state.bio
    };

    this.props.registerUser(newUser, this.props.history);
  };

  render() {
    const { errors } = this.props;
    return (
      <>
        <Navbar />
        <div className="container">
          <div className="row mt-5">
            <div className="col-md-1 col-lg-2" />
            <div className="col-md-10 col-lg-8 card mt-5 text-center p-0 entry">
              <div className="card-body p-5">
                <h3 className="mb-5">Sign up</h3>
                <form noValidate onSubmit={this.onSubmit}>
                  <div className="row">
                    <div className="col-lg-6">
                      <div className="mb-4">
                        <InputGroup
                          id="name"
                          placeholder="Name"
                          name="name"
                          value={this.state.name}
                          onChange={this.onChange}
                          error={errors.name}
                        />
                      </div>
                      <div className="mb-4">
                        <InputGroup
                          id="email"
                          placeholder="Email"
                          name="email"
                          value={this.state.email}
                          onChange={this.onChange}
                          error={errors.email}
                        />
                      </div>
                      <div className="mb-4">
                        <InputGroup
                          id="password"
                          placeholder="Password"
                          name="password"
                          type="password"
                          value={this.state.password}
                          onChange={this.onChange}
                          error={errors.password}
                        />
                      </div>
                      <InputGroup
                        id="password2"
                        placeholder="Confirm Password"
                        name="password2"
                        type="password"
                        value={this.state.password2}
                        onChange={this.onChange}
                        error={errors.password2}
                      />
                    </div>
                    <div className="col-lg-6">
                      <div className="mb-4">
                        <InputGroup
                          id="location"
                          placeholder="Location"
                          name="location"
                          value={this.state.location}
                          onChange={this.onChange}
                          error={errors.location}
                        />
                      </div>
                      <TextFieldGroup
                        id="bio"
                        placeholder="Bio"
                        name="bio"
                        value={this.state.bio}
                        onChange={this.onChange}
                        error={errors.bio}
                      />
                    </div>
                  </div>

                  <button type="submit" className="btn btn-dark btn-block mt-4">
                    Create Account
                  </button>
                </form>
              </div>
              <div className="text-muted card-footer mt-3">
                Already have an account? <Link to="/login">Log in</Link>
              </div>
            </div>
          </div>
        </div>
      </>
    );
  }
}

Register.propTypes = {
  registerUser: PropTypes.func.isRequired,
  errors: PropTypes.object.isRequired
};

export default connect(
  null,
  { registerUser }
)(redirectAuthenticated(handleInputErrors(Register)));
