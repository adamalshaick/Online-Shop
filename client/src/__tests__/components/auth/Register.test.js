import React from "react";
import { Register } from "../../../components/auth/Register";
import { shallow } from "enzyme";

const mockRegisterfn = jest.fn();
const Props = {
  auth: {
    isAuthenticated: false
  },
  errors: {}
};
let wrapper;

wrapper = shallow(<Register {...Props} registerUser={mockRegisterfn} />);

it("displays form", () => {
  expect(wrapper.find("form").length).toEqual(1);
});

it("should call the mock register function", () => {
  wrapper.find("form").simulate("submit", { preventDefault() {} });
  expect(mockRegisterfn.mock.calls.length).toBe(1);
});

describe("register action", () => {
  wrapper
    .find("#name")
    .simulate("change", { target: { name: "name", value: "testName" } });

  wrapper
    .find("#email")
    .simulate("change", { target: { name: "email", value: "test@gmail.com" } });

  wrapper.find("#password").simulate("change", {
    target: { name: "password", value: "passwordTest" }
  });

  wrapper.find("#password2").simulate("change", {
    target: { name: "password2", value: "password2Test" }
  });

  it("calls function with correct data", () => {
    expect(mockRegisterfn.mock.calls[0][0]).toEqual({
      name: "testName",
      email: "test@gmail.com",
      password: "passwordTest",
      password2: "password2Test"
    });
  });
});
