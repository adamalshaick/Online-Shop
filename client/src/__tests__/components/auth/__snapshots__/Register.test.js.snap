// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<Fragment>
  <Connect(Navbar) />
  <div
    className="container"
  >
    <div
      className="row mt-5"
    >
      <div
        className="col-md-2 col-lg-3"
      />
      <div
        className="col-md-8 col-lg-6 card mt-5 text-center p-0 entry"
      >
        <div
          className="card-body p-5"
        >
          <h3
            className="mb-5"
          >
            Sign up
          </h3>
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <TextFieldGroup
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="text"
              value="testName"
            />
            <TextFieldGroup
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value="test@gmail.com"
            />
            <TextFieldGroup
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value="passwordTest"
            />
            <TextFieldGroup
              id="password2"
              name="password2"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value="password2Test"
            />
            <button
              className="btn btn-dark btn-block mt-4"
              type="submit"
            >
              Create Account
            </button>
          </form>
        </div>
        <div
          className="text-muted card-footer mt-3"
        >
          Already have an account? 
          <Link
            replace={false}
            to="/login"
          >
            Log in
          </Link>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
